internal/log/log_test.go

```diff
@@ -0,0 +1,153 @@
+package log
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"testing"
+)
+
+func TestSetLevel(t *testing.T) {
+	// Check default level is INFO & Is* flags are set appropriately
+	if !(logLevel == INFO && IsError == true && IsWarn == true && IsInfo == true && IsDebug == false &&
```

Comment:
For boolean value no need to do the `var == true` just call the `var`. i.e:
```
if level != TRACE || !IsError || !IsWarn || !IsInfo || !IsDebug || !IsTrace {
      t.Errorf("TRACE level is not set as expected.")
}
```

I would go as far as making a helper function:
```
func levelChecksOut(lvl Level)(ok bool) {
    if level != lvl  {
        return false
   }
   switch lvl {

      case TRACE:
           if !IsTRACE {
                 return false
           }
           fallthrough
      case DEBUG:
           if !IsDEBUG {
                return false
          }
           fallthrough
      default: 
        case INFO:
          if !IsINFO {
                 return false
          }
           fallthrough
       case WARN:
           if !IsWARN {
                 return false
           }
           fallthrough
         case ERROR:
            if !IsERROR {
                  return false
           }
           fallthrough
         case FATAL:
          if !IsFATAL {
                return false
          }
           fallthrough
       }
     return true
}
```
The in the test function:
```
if levelChecksOut(TRACE) {
      t.Errorf("TRACE level is not set as expected.")
}
```